enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,973 @@
+  * Control-plane components should keep working without any issues.
+    * Users should be e.g. able to delete PVCs and expect that the corresponding volume is deleted in AWS.
+    * Dynamic provisioning should work if forced by `volumeBindingMode: Immediate`. Users are then responsible
+      for creating nodes in the right availability zones so that they can use the provisioned volumes.
+
+* Phase 2: As a HyperShift user, I want to get metrics of storage management cluster components so I can monitor them.
+  * This is Phase 2, as most useful storage metrics are already emitted by kube-controller-manager. While metrics of the
+    CSI driver, its sidecars, snapshot-controller and their operators are interesting, they're not crucial.
+
+* Phase 2: As a HyperShift user, I want all management cluster components that talk to a guest cluster to have proper
+  RBACs, so a compromised storage component can't break the guest cluster (too much).
+  * The initial implementation may give cluster-admin role in the guest cluster to storage components in the management
+    cluster - a compromised component in the management cluster could get full access the guest cluster, but not to the
+    management cluster.
+  * Even with the minimal RBACs, CSI driver components will be able to delete any PV and any volume in the cloud,
+    messing up the guest cluster pretty badly.
+
+### Goals
+
+Move storage control plane components to HyperShift management cluster.
+
+### Non-Goals
+
+In Phase 1:
+
+* Metrics of the storage components running in the management cluster.
+* Fine-grained guest cluster RBACs for the storage components running in the management cluster.
+
+Overall:
+
+* OLM based CSI driver(s).
+  * For example, AWS EFS CSI driver will be installed by OLM as usual, and it will run in the guest cluster only.
+    There is no plan to move its controller parts to the management cluster.
+  * Similarly, local-storage-operator will run in the guest cluster only.
+
+## Proposal
+
+Run following storage components in a **HyperShift management cluster**:
+
+* cluster-storage-operator
+* aws-ebs-csi-driver-operator
+* AWS EBS CSI driver controllers
+
+* cluster-csi-snapshot-controller-operator
+* csi-snapshot-controller
+
+In the **guest cluster** remains:
+
+* AWS EBS CSI driver DaemonSet - it must format + mount volumes on worker nodes.
+* All storage objects - CSIDrivers, StorageClasses, VolumeSnapshotClasses, PersistentVolumes, PersistentVolumeClaims,
+  VolumeAttachments, ...
+
+All operators mentioned above will basically get two kubeconfigs (one to the management cluster and one to the guest
+cluster) and they will create their operands in the right clusters.
+
+### Workflow Description
+
+Generic interaction between storage operators should be:
+
+1. User creates HostedCluster in Hypershift.
+2. control-plane-operator creates the necessary cluster-storage-operator abd cluster-csi-snapshot-controller-operator
```

Comment:
nit: clarify it not only creates, also reconciles i.e manages the lifecycle.

Reply:
Fixed